dependencies.yaml

```diff
@@ -88,6 +97,17 @@ channels:
   - nvidia
 dependencies:
   build:
+    common:
+      - output_types: [conda]
+        packages:
+          - scikit-build-core>=0.7.0
+      - output_types: [requirements, pyproject]
+        packages:
+          - scikit-build-core[pyproject]>=0.7.0
+      - output_types: [requirements, pyproject, conda]
+        packages:
+          - rapids-build-backend>=0.3.0,<0.4.0.dev0
```

Comment:
For other RBB PRs, we've used an order more like:
- rapids-build-backend (conda, requirements, pyproject)
- scikit-build-core (conda)
- scikit-build-core (requirements, pyproject)